### PR TITLE
Added unit tests for GraphCanvas translation system

### PR DESCRIPTION
**Pull request by lsemp3d**
Reorganized the code to enable testing of GraphCanvas static library code and added tests for the translation system.

Signed-off-by: lsemp3d <58790905+lsemp3d@users.noreply.github.com>

### CODE REVIEW

**Review by carlitosan — APPROVED** [review]

File: Gems/GraphCanvas/Code/CMakeLists.txt [hunk 1]
```diff
@@ -78,3 +80,35 @@ if (PAL_TRAIT_BUILD_HOST_TOOLS)
...
+if(PAL_TRAIT_BUILD_TESTS_SUPPORTED)
+
+    ly_add_target(
+        NAME GraphCanvas.Tests ${PAL_TRAIT_TEST_TARGET_TYPE}
+        NAMESPACE Gem
+        FILES_CMAKE
+            graphcanvas_tests_files.cmake
+        COMPILE_DEFINITIONS
+            PRIVATE
+                GRAPHCANVAS_TESTS
```
Comment (onecent1101) [inline Gems/GraphCanvas/Code/CMakeLists.txt:96]: is this required? don't see it get used

File: Gems/GraphCanvas/Code/CMakeLists.txt [hunk 2]
```diff
@@ -78,3 +80,35 @@ if (PAL_TRAIT_BUILD_HOST_TOOLS)
...
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Tests
+                Source
+        BUILD_DEPENDENCIES
+            PRIVATE
+                AZ::AzTest
+                Gem::GraphCanvas
```
Comment (onecent1101) [inline Gems/GraphCanvas/Code/CMakeLists.txt:104]: Seems like you only need to include Gem::GraphCanvasWidgets as build dependency, so what is Gem::GraphCanvas used for? and I don't think you need runtime dependency for your unit test (which is used to bring and copy shared libs)?

**Review by onecent1101 — COMMENTED** [review]
Nit: considering rename Gems/GraphCanvas/Code/Tests/GraphCanvasTranslation.cpp to Gems/GraphCanvas/Code/Tests/TranslationBusTest.cpp, so it will be easier to locate which file it is testing.

File: Gems/GraphCanvas/Code/CMakeLists.txt (hunk 1 above)
Comment (lsemp3d) [reply Gems/GraphCanvas/Code/CMakeLists.txt:96]: Seemed fairly standard among the examples I saw, I could remove it

File: Gems/GraphCanvas/Code/CMakeLists.txt [hunk 3]
```diff
@@ -6,6 +6,8 @@
 #
 #
 
+if(PAL_TRAIT_BUILD_HOST_TOOLS)
```
Comment (onecent1101) [inline Gems/GraphCanvas/Code/CMakeLists.txt:9]: in general, this flag is for tool and builder variant? unless this gem is purely a tool/builder itself?

**Review by onecent1101 — APPROVED** [review]

File: Gems/GraphCanvas/Code/CMakeLists.txt (hunk 3 above)
Comment (lsemp3d) [reply Gems/GraphCanvas/Code/CMakeLists.txt:9]: GraphCanvas is a tool framework, it does not have or need a runtime equivalent

**Review by cgalvan — APPROVED** [review]

**Pull request merged**